mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.multi
+    @api.returns('mail.message', lambda value: value.id)
+    def message_post(
+        self, body='', subject=None,
+        message_type='notification', subtype=None,
+        parent_id=False, attachments=None,
+        notif_layout=False, add_sign=True, model_description=False,
+        mail_auto_delete=True, **kwargs
+    ):
+        subtype_id = kwargs.get('subtype_id', False)
+        if not subtype_id:
+            subtype = subtype or 'mt_note'
+            if '.' not in subtype:
+                subtype = 'mail.%s' % subtype
+            subtype_id = self.env['ir.model.data'].xmlid_to_res_id(subtype)
+        mail_template = self.env["mail.message.custom.subject"].search([
+            ("model_id.model", "=", self._name),
+            ("subtype_ids", "=", subtype_id),
+        ])
+        subject_override = mail_template.render_subject_template(
+            self._name, self.id
+        )
+        subject = subject_override or subject
```

Comment:
```suggestion
        if mail_template:
            subject_override = mail_template.render_subject_template(
                self._name, self.id
            )
            subject = subject_override or subject
```

Reply:
Thank you! :+1: 
I don't think it would cause problems, though, as the search would return an empty recordset, which [will then be iterated over](https://github.com/OCA/social/pull/650/files#diff-9c0df1d72822413b9e7331e329dd186ca0093c700694cabe007f8def001bf5baR172), right?
Just a curiosity :smile: Change made :+1: